modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/rebalance/RebalanceMinimumRequiredTimeProviderImpl.java

```diff
@@ -172,9 +156,9 @@ Map<Integer, NavigableMap<Long, CatalogZoneDescriptor>> allZonesByTimestamp(
                 NavigableMap<Long, CatalogZoneDescriptor> map = allZones.computeIfAbsent(zone.id(), id -> new TreeMap<>());
 
                 if (map.isEmpty() || updateRequiresAssignmentsRecalculation(map.lastEntry().getValue(), zone)) {
-                    map.put(catalog.time(), zone);
+                    map.put(zone.updateTimestamp().longValue(), zone);
```

Comment:
Can we use `HybridTimestamp` here for keys? This will make code more readable. This class is `Comparable`, so no worries about that. Please update all the usages in this class, thank you!

Reply:
done